packages/lucide/src/lucide.ts

```diff
@@ -22,11 +32,16 @@ const createIcons = ({
   }
 
   const elementsToReplace = Array.from(root.querySelectorAll(`[${nameAttr}]`));
-  const templates = Array.from(root.querySelectorAll('template'));
-  templates.forEach((template) => {
-    const contentToReplace = Array.from(template.content.querySelectorAll(`[${nameAttr}]`));
-    elementsToReplace.push.apply(elementsToReplace, contentToReplace);
-  });
+
+  if (replaceInsideTemplates) {
+    const templates = Array.from(root.querySelectorAll('template'));
+
+    templates.forEach((template) => {
+      const contentToReplace = Array.from(template.content.querySelectorAll(`[${nameAttr}]`));
+      elementsToReplace.push.apply(elementsToReplace, contentToReplace);
+    });
```

Comment:
@ericfennis I wonder if maybe this feature should be made recursive? 🤔

In essence, something like:

```ts
if (replaceInsideTemplates) {
  Array.from(root.querySelectorAll('template')).forEach(template => createIcons({
    icons,
    nameAttr,
    attrs,
    root: template,
    replaceInsideTemplates,
  });
}

Reply:
Good one, this makes the code a lot cleaner.

Reply:
True, but it is also nice to extend the current API to enable this.